src/setup/rgw.py

```diff
@@ -14,77 +14,219 @@
 
+    _machine_id = Conf.machine_id
+
     @staticmethod
     def validate(phase: str):
-        """Perform validtions."""
+        """Perform validations."""
+
+        Log.info(f'validations started for {phase} phase.')
+
+        if phase == 'post_install':
+            # Perform RPM validations
+            for rpms in [CORTX_RPMS, CEPH_RPMS]:
+                PkgV().validate('rpms', rpms)
+            Log.info(f'All required rpms are installed on {Rgw._machine_id} node.')
+        elif phase == 'prepare':
+            Rgw._file_exist(CEPH_CONF_TMPL)
+        elif phase == 'config':
+            Rgw._file_exist(CEPH_CONF)
+
+        Log.info(f'validations completed for {phase} phase.')
 
-        # Perform RPM validations
         return 0
 
     @staticmethod
-    def post_install(config_path: str):
-        """
-        Performs post install operations.
-        """
+    def post_install(conf: MappedConf):
+        """Performs post install operations."""
 
+        Log.info('PostInstall phase completed.')
         return 0
 
     @staticmethod
-    def prepare(config_path: str):
-        """
-        Performs prepare phase operations.
-        """
+    def prepare(conf: MappedConf):
+        """Perform prepare operations."""
+
+        Log.info('Prepare phase started.')
+
+        try:
+            hostname = conf.get(f'node>{Rgw._machine_id}>hostname')
+            host_ip = socket.gethostbyname(hostname)
+            ceph_tmpl_idx = 'ceph_conf_tmpl'
+            ceph_tmpl_url = f'ini://{CEPH_CONF_TMPL}'
+            # Load template ceph.conf file and updated required field.
+            kv_list = [
+                ('global>fsid', str(uuid.uuid1())),
+                ('global>mon host', host_ip)]
+            Rgw._update_config(ceph_tmpl_idx, ceph_tmpl_url, kv_list)
+
+            # Copy ceph_conf_tmpl_file to /etc/ceph/ dir.
+            shutil.copyfile(CEPH_CONF_TMPL, CEPH_CONF)
+            Log.info(f'{CEPH_CONF_TMPL} config copied to {CEPH_CONF}.')
+
+            # create admin keyring.
+            cmd = f"sudo ceph-authtool --create-keyring {CEPH_ADMIN_KEYRING} \
+                --gen-key -n client.admin --cap mon 'allow *' --cap osd 'allow *'"
+            _, err, rc, = SimpleProcess(cmd).run()
+            if rc != 0:
+                raise SetupError(errno.EINVAL, f'"{cmd}" failed with error {err}.')
+        except Exception as e:
+            raise SetupError(errno.EINVAL, f'Error ocurred while fetching node ip, {e}')
+
+        Log.info('Prepare phase completed.')
 
         return 0
 
     @staticmethod
-    def config(config_path: str):
-        """
-        Performs configurations.
-        """
+    def config(conf: MappedConf):
+        """Performs configurations."""
+
+        Log.info('Config phase started.')
 
+        # TODO: Inside [client.rgw.<hostname>] section and
+        # add motr,hax endpoints and rgw client related parameters.
+        Log.info('Config phase completed.')
```

Comment:
As per https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/862126560/RGW+Deployment#RGW-Mini-Provisioner-Interfaces , following we need to do right. Are you still in process of implementing it

> Validate sysconfig file.
Update config files of RGW based on Gconfstore values.
Update motr,hax  endpoints and rgw client related parameters into ceph config([client.rgw.<hostname>] section.)

Reply:
Got it